packages/prompts/src/index.ts

```diff
@@ -1014,3 +1014,482 @@ export const tasks = async (tasks: Task[], opts?: CommonOptions) => {
+			cursor = filtered.length - 1;
+		}
+		
+		// Update selected value based on cursor
+		if (filtered.length > 0) {
+			selectedValue = filtered[cursor].value;
+		}
+	};
+	
+	// Create text prompt
+	const prompt = new TextPrompt({
+		placeholder: opts.placeholder,
+		initialValue: '',
+		input: opts.input,
+		output: opts.output,
+		render() {
+			// Register event handlers only once
+			if (!isEventsRegistered) {
+				// Handle keyboard navigation
+				this.on('key', (key) => {
+					// Start navigation mode with up/down arrows
+					if (key === 'up' || key === 'down') {
+						isNavigating = true;
+					}
+					
+					// Allow typing again when user presses any other key
+					if (key !== 'up' && key !== 'down' && key !== 'return') {
+						isNavigating = false;
+					}
+				});
+				
+				// Handle cursor movement
+				this.on('cursor', (key) => {
+					if (filtered.length === 0) return;
+					
+					// Enter navigation mode
+					isNavigating = true;
+					
+					if (key === 'up') {
+						cursor = cursor > 0 ? cursor - 1 : filtered.length - 1;
+					} else if (key === 'down') {
+						cursor = cursor < filtered.length - 1 ? cursor + 1 : 0;
+					}
+					
+					// Update selected value
+					if (filtered.length > 0) {
+						selectedValue = filtered[cursor].value;
+					}
+				});
+				
+				// Register input change handler to update filtering
+				this.on('value', () => {
+					// Only update input when not in navigation mode
+					if (!isNavigating) {
+						const newInput = this.value || '';
+						if (newInput !== input) {
+							input = newInput;
+							filterOptions(input);
+						}
+					}
+				});
+				
+				isEventsRegistered = true;
+			}
+			
+			// Handle initial state
+			if (this.state === 'initial') {
+				input = this.value || '';
+				filterOptions(input);
+				
+				// Set initial selection if provided
+				if (opts.initialValue !== undefined && !selectedValue) {
+					const initialIndex = opts.options.findIndex(o => o.value === opts.initialValue);
+					if (initialIndex !== -1) {
+						cursor = initialIndex;
+						selectedValue = opts.options[initialIndex].value;
+					}
+				}
+			}
+			
+			// Set selection on submit
+			if (this.state === 'submit') {
+				this.value = selectedValue as any;
+			}
+			
+			// Title and message display
+			const title = `${color.gray(S_BAR)}\n${symbol(this.state)}  ${opts.message}\n`;
+			
+			// Handle different states
+			switch (this.state) {
+				case 'submit': {
+					// Show selected value
+					const selected = opts.options.find(o => o.value === selectedValue);
+					const label = selected ? getLabel(selected) : '';
+					return `${title}${color.gray(S_BAR)}  ${color.dim(label)}`;
+				}
+				
+				case 'cancel': {
+					return `${title}${color.gray(S_BAR)}  ${color.strikethrough(color.dim(this.value || ''))}`;
+				}
+				
+				default: {
+					// Mode indicator (for debugging)
+					const modeIndicator = isNavigating 
+						? color.yellow(' [navigation]') 
+						: '';
```

Comment:
Maybe we could highlight which mode is active in the footer? `up/down to select` could be white when someone is navigating, `type to search` could be white when someone is in filter mode